src/components/coffeechat.ts

```diff
@@ -0,0 +1,276 @@
+    `SELECT * FROM coffee_future_matches WHERE week_id = ${week_id};`
+  )) as future_match[]).map((entry) => [entry.first_user_id, entry.second_user_id]);
+  await db.run(`UPDATE coffee_week_status SET finished = 1 WHERE week_id = ${week_id};`);
+  const activeMatchIDs = [...(await loadNotMatched(client)).keys()];
+  let single: string | null = null;
+  if (activeMatchIDs.length % 2 == 1) {
+    const usedIDs = _.flatten(matches);
+    for (const key of activeMatchIDs) {
+      if (!usedIDs.includes(key)) {
+        single = key;
+        break;
+      }
+    }
+  }
+  return { matches: matches, single: single };
+};
+
+const loadNotMatched = async (client: Client): Promise<Map<string, number>> => {
+  const userList = (await (await client.guilds.fetch(TARGET_GUILD_ID)).members.fetch())
+    ?.filter((member) => member.roles.cache.has(COFFEE_ROLE_ID))
+    .map((member) => member.user.id);
+  const notMatched: Map<string, number> = new Map();
+  userList.forEach((val: string, index: number) => {
+    notMatched.set(val, index);
+  });
+  return notMatched;
+};
+
+const hasDupe = (matched: number[][], matches: string[][], userList: Map<string, number>): boolean => {
+  for (const [personA, personB] of matches) {
+    if (matched[userList.get(personA)!][userList.get(personB)!] > 1) return true;
+  }
+  return false;
+};
+
+const getMaxDupe = (matched: number[][], matches: string[][], userList: Map<string, number>): number => {
+  let maxDupe = 0;
+  for (const [personA, personB] of matches) {
+    maxDupe = Math.max(maxDupe, matched[userList.get(personA)!][userList.get(personB)!]);
+  }
+  return maxDupe;
+};
+
+const loadMatched = async (notMatched: Map<string, number>): Promise<number[][]> => {
+  const db = await openDB();
+  const matched: number[][] = new Array(notMatched.size).fill(0).map(() => new Array(notMatched.size).fill(0));
+  const matches = (await db.all(`SELECT * FROM coffee_historic_matches`)) as historic_match[];
+  for (const { first_user_id, second_user_id } of matches) {
+    if (notMatched.has(first_user_id) && notMatched.has(second_user_id)) {
+      matched[notMatched.get(first_user_id)!][notMatched.get(second_user_id)!]++;
+      matched[notMatched.get(second_user_id)!][notMatched.get(first_user_id)!]++;
+    }
+  }
+  return matched;
+};
+
+export const writeNewMatches = async (newMatches: string[][], date: Date): Promise<void> => {
+  const db = await openDB();
+
+  //leverages datetime() in sqlite to convert to some standard of sqlite time
+  await db.run(
+    `INSERT INTO coffee_historic_matches (first_user_id, second_user_id, date) VALUES ${_.join(
+      newMatches.map((entry) => `('${entry[0]}', '${entry[1]}', datetime('${date.toISOString()}'))`),
+      ','
+    )};`
+  );
+};
+
+const wipeHistory = async (dbName: string): Promise<void> => {
+  const db = await openDB();
+  await db.run(`DELETE FROM ${dbName}`);
+};
+
+const stableMatch = (userList: Map<string, number>, matched: number[][]): string[][] => {
```

Comment:
We need better function documentation and comments for all functions in this file.

Reply:
mhm, im a little lost